rllib/algorithms/algorithm_config.py

```diff
@@ -2532,17 +2538,45 @@ def offline_data(
+                when sampling from the dataset. This is needed and defaults to the
+                `ray.rllib.utils.replay_buffers.EpisodeReplayBuffer` in case of a
+                dataset storing RLlib's old API stack's `SampleBatch`es, i.e.
+                `input_read_sample_batches=True` or a dataset storing new API stack
+                `SingleAgentEpisode`s, i.e. `input_read_episodes=True`. As each
+                `SampleBatch` or `SingleAGentEpisode` could potentially hold more than a
+                single timestep, the episode buffer is used to buffer episodes and
+                sample exactly the defined `train_batch_size_per_learner` from it. Any
+                buffer that inherits from
+                `ray.rllib.utils.replay_buffers.EpisodeReplayBuffer` can be used, e.g.
+                you could use RLlib's prrioritized episode buffer
+                `ray.rllib.utils.replay_buffers.PrioritizedEpsiodeReplayBuffer`. See
+                `prelearner_buffer_kwargs` for passing in further arguments into the
+                buffer's constructor. Note, the buffer acts (a) as a buffer to dissect
+                `SampleBatch` data into smaller chunks, (b) to concatenate epiosde
+                chunks from different `SampleBatch`es, and (c) to act a shuffler that
+                shuffles experiences from different episodes.
+            prelearner_buffer_kwargs: `kwargs` for the `prelearner_buffer_class`. If no
+                arguments are passed in the buffer's `capacity` will be set to `10 x
+                train_batch_size_per_learner` and the `batch_size_B` to the
+                `train_batch_size_per_learner`. The capacity will act (a) as a shuffle
+                for experiences randomizing experiences while sampling and (b) as a
+                buffer for the episodes, i.e. will keep episodes longer available for
+                sampling depending on the capacity. If you want to traverse the dataset
+                as is and avoid sampling the same experiences multiple times choose a
+                small enough capacity to store episodes from your `SampleBatch` data,
+                but to evict old experiences as soon as possible.
             prelearner_module_synch_period: The period (number of batches converted)
                 after which the `RLModule` held by the `PreLearner` should sync weights.
                 The `PreLearner` is used to preprocess batches for the learners. The
                 higher this value the more off-policy the `PreLearner`'s module will be.
                 Values too small will force the `PreLearner` to sync more frequently
                 and thus might slow down the data pipeline. The default value chosen
                 by the `OfflinePreLearner` is 10.
-            dataset_num_iters_per_learner: Number of iterations to run in each learner
+            dataset_num_iters_per_learner: Number of updates to run in each learner
```

Comment:
👍

Reply:
I admit, `dataset_num_iters_per_learner` is not straight to the point here. `dataset_num_updates_per_learner` is not better in my opinion. Even though `dataset_num_batches_per_learner` describes better that these many different! batches are pulled per learner, but does not point out that it is actually deeply related to an `DataIterator` that iterates these many times. I am not sure, yet. As long as I am not sure, I will leave it as is ;)